aws-datastore/src/main/java/com/amplifyframework/datastore/DataStoreConfiguration.java

```diff
@@ -58,6 +70,76 @@ public static Builder builder() {
         return new Builder();
     }
 
+    /**
+     *
+     * @param context application context which will be used to retrieve amplicationconfiguration.json
+     * @param pluginConfigKey the desired dataStore plugin config key (i.e. awsDataStorePlugin)
+     * @return A new builder instance
+     * @throws DataStoreException
+     */
+    @NonNull
+    public static Builder builder(@NonNull Context context, @NonNull String pluginConfigKey) throws DataStoreException {
+        Objects.requireNonNull(context);
+        Objects.requireNonNull(pluginConfigKey);
+        JSONObject pluginConfig = null;
+        try {
+            pluginConfig = Objects.requireNonNull(AmplifyConfiguration.fromConfigFile(context)
+                .forCategoryType(CategoryType.DATASTORE))
+                .getPluginConfig(pluginConfigKey);
+        } catch (AmplifyException | NullPointerException exception) {
+            Log.w(TAG, "Unable to read DataStore configuration from file.", exception);
```

Comment:
I'm a little unclear on our approach to logging. Should this use a unified logger, a la:

```java
private static final Logger LOG = Amplify.Logging.forNamespace("amplify:aws-datastore);
```